multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/security/AuthorizationChecker.java

```diff
@@ -94,8 +95,12 @@ boolean checkPermissions(UserInfo userInfo, String spaceId, boolean readOnly) {
         if (hasAdminScope(userInfo)) {
             return true;
         }
-
-        UUID spaceGuid = UUID.fromString(spaceId);
+        UUID spaceGuid = null;
+        try {
+            spaceGuid = UUID.fromString(spaceId);
+        } catch (IllegalArgumentException e) {
+            throw new NotFoundException(e.getMessage(), e);
+        }
```

Comment:
I would suggest extracting these lines of code in separate method, because try-catch blocks are extremely ugly.
Consider something like the following example:
```
private UUID convertToUUID(String spaceId) { // you may come with better name
    try {
            return UUID.fromString(spaceId);
     } catch (IllegalArgumentException e) {
            throw new NotFoundException(e.getMessage(), e); // this exception will propagate in case it is thrown
     }
}
```